### PR TITLE
Generate newsletter module

### LINKED ISSUE #5: generate-newsletter module
TODO:

- [x] Move it same repo as newsletter.
- [x] - CLI command to generate issue. 
- [x] Update the template contents.

### PR DESCRIPTION
**Pull request by shidhincr**
Fixes #5 

[asciicast](https://asciinema.org/a/PCpmGnhbUjCFqlxo6LrwKY2nF)

### CODE REVIEW

File: packages/generate-newsletter/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	</table>
+</center>	
+		`
+    const currentYear = userResponse.date.split('-').shift()
+    const month = userResponse.month.substring(0, 3).toLowerCase()
+    const issueNo = userResponse.issue
+    // @TODO: update the content title
+    const markdown = `---
+path: "/issues/${currentYear}/${month}/${issueNo}"
+published: true
```
Comment (gokulkrishh) [inline packages/generate-newsletter/index.js]: can we make it as false. We should manually enable this when we are publishing newsletter every time. It sort of a check for us to verify everything before pushing.

File: packages/generate-newsletter/index.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+date: "${userResponse.date}"
+title: "Issue #${issueNo}"
+contentTitle: Run <code>copy(obj)</code> in the Chrome Dev Tools console to copy an object to your clipboard.
```
Comment (gokulkrishh) [inline packages/generate-newsletter/index.js]: Can we make it empty ?

File: packages/generate-newsletter/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+<center>
+	<table align="center" border="0" cellspacing="0" width="100%" height="100%" cellpadding="0">
+	<tbody>${links}</tbody>
+	</table>
+</center>	
+		`
+    const currentYear = userResponse.date.split('-').shift()
+    const month = userResponse.month.substring(0, 3).toLowerCase()
+    const issueNo = userResponse.issue
+    // @TODO: update the content title
```
Comment (gokulkrishh) [inline packages/generate-newsletter/index.js]: 👍

File: packages/generate-newsletter/index.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+const fs = require('fs-extra')
+const path = require('path')
+const htmlCreator = require('html-creator')
+const { prompt } = require('enquirer')
+const { extractSheets } = require('spreadsheet-to-json')
+const argv = require('yargs').argv
+
+const { getRandomColor, colorLuminance, getCurrentDate } = require('./utils')
+const credentials = require('./credentials.json')
```
Comment (gokulkrishh) [inline packages/generate-newsletter/index.js]: can we add a empty file?

File: packages/generate-newsletter/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	"license": "MIT",
+	"devDependencies": {
+		"dotenv": "^6.2.0",
+		"html-creator": "https://github.com/gokulkrishh/html-creator.git",
+		"spreadsheet-to-json": "^1.3.1"
+	},
+	"dependencies": {
+		"enquirer": "^2.3.0",
+		"fs-extra": "^7.0.1",
+		"yargs": "^13.1.0"
```
Comment (gokulkrishh) [inline packages/generate-newsletter/package.json]: 🔥

File: packages/generate-newsletter/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	"description": "Generate newsletter template via google sheets API",
+	"main": "index.js",
+	"scripts": {
+		"start": "node index.js"
+	},
+	"bin": {
+		"generate-newsletter": "cli.js"
+	},
+	"repository": "https://github.com/gokulkrishh/generate-newsletter",
+	"author": "Gokulakrishnan Kalaikovan <krishnangokul9@gmail.com>",
```
Comment (gokulkrishh) [inline packages/generate-newsletter/package.json]: Change the author ?

File: packages/generate-newsletter/utils.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  if (mm < 10) {
+    mm = '0' + mm
+  }
+
+  return yyyy + '-' + mm + '-' + dd
+}
+
+const colorLuminance = (hex, lum) => {
```
Comment (gokulkrishh) [inline packages/generate-newsletter/utils.js]: We are not using this anymore, Lets remove this ?

File: packages/generate-newsletter/utils.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    c = Math.round(Math.min(Math.max(0, c + c * lum), 255)).toString(16)
+    rgb += ('00' + c).substr(c.length)
+  }
+
+  return rgb
+}
+
+module.exports = {
+  getRandomColor,
+  colorLuminance,
```
Comment (gokulkrishh) [inline packages/generate-newsletter/utils.js]: Here as well

File: packages/generate-newsletter/index.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
+const fs = require('fs-extra')
+const path = require('path')
+const htmlCreator = require('html-creator')
+const { prompt } = require('enquirer')
+const { extractSheets } = require('spreadsheet-to-json')
+const argv = require('yargs').argv
+
+const { getRandomColor, colorLuminance, getCurrentDate } = require('./utils')
```
Comment (gokulkrishh) [inline packages/generate-newsletter/index.js]: Remove here as well?

File: packages/generate-newsletter/index.js (hunk 1 above)
Comment (shidhincr) [reply packages/generate-newsletter/index.js]: 👍

File: packages/generate-newsletter/index.js (hunk 2 above)
Comment (shidhincr) [reply packages/generate-newsletter/index.js]: yeah.. we can make it empty now. But how exactly we should populate this one?

File: packages/generate-newsletter/index.js (hunk 4 above)
Comment (shidhincr) [reply packages/generate-newsletter/index.js]: yeah.. but the file is being added to gitignore.. so how we should do it?

File: packages/generate-newsletter/package.json (hunk 6 above)
Comment (shidhincr) [reply packages/generate-newsletter/package.json]: keep this like this :)

File: packages/generate-newsletter/utils.js (hunk 7 above)
Comment (shidhincr) [reply packages/generate-newsletter/utils.js]: yeah.. these should be removed. I will make those changes

File: packages/generate-newsletter/index.js (hunk 9 above)
Comment (shidhincr) [reply packages/generate-newsletter/index.js]: 👍

File: packages/generate-newsletter/index.js (hunk 4 above)
Comment (shidhincr) [reply packages/generate-newsletter/index.js]: Added a different condition to check this one.

File: packages/generate-newsletter/index.js (hunk 2 above)
Comment (gokulkrishh) [reply packages/generate-newsletter/index.js]: I think this is something we have to manually add it. Title should be something interesting topic among the links ?

File: packages/generate-newsletter/index.js (hunk 4 above)
Comment (gokulkrishh) [reply packages/generate-newsletter/index.js]: Aww. ya lets leave it then. We can update in readme. Things needed to generate newsletter

File: packages/generate-newsletter/index.js (hunk 2 above)
Comment (shidhincr) [reply packages/generate-newsletter/index.js]: 👍

**Comment by shidhincr** [comment]
@gokulkrishh I am not able to see why the build is failed. Can you give me access?

**Comment by gokulkrishh** [comment]
@shidhincr Sure.

**Comment by gokulkrishh** [comment]
@shidhincr Check now. Logs are public.

**Pull request merged**